internal/engine/upper.go

```diff
@@ -502,13 +503,7 @@ func handleServiceEvent(ctx context.Context, state *store.EngineState, service *
 		return
 	}
 
-	url, err := k8s.ServiceURL(service)
-	if err != nil {
-		logger.Get(ctx).Infof("error resolving service %s: %v", manifestName, err)
-		return
-	}
-
-	ms.LBs[k8s.ServiceName(service.Name)] = url
+	ms.LBs[k8s.ServiceName(service.Name)] = action.URL
```

Comment:
maybe only add this if url is non-nil?

Reply:
nope! this is necessary for correctness. for example, if you edit the service yaml

Reply:
added a unit test for this case

Reply:
ah gotcha! Sure. (and i realize elsewhere that we check if the elem in here is nil before adding it to []endpoints so 👌 )